### DISCUSSION TITLE
I have question for  my doubts.

### DISCUSSION THREAD

**Discussion by dagi0627** [Beormund/Tasmota32-Multi-Zone-Heating-Controller#14] [General]
It starts 384 row
```     
def set_target_temp(temp)
        # Only use a schedule target temp if mode is Auto or Day
        var auto_or_day = 33 & (1<<self.get_mode())
        if auto_or_day && self.get_power() && temp
            self.target_temp = temp
        else
            self.target_temp = self.get_temp(zone.target)
        end
    end
```
In above code have same result of if condition. These are  
```
self.target_temp = temp
```
and 
``` 
self.target_temp = self.get_temp(zone.target)
```
same result. 
If it is maybe my fault, these rows code what purpose?

**Comment by Beormund** [comment]
If temp is passed into set_target_temp() then that value is used for the target temperature otherwise the zone's target temperature is used instead.

**Reply by dagi0627** [reply]
If you look closely, you're putting a same value into a variable self.target_temp
![Screenshot from 2023-01-31 14-20-07](https://user-images.githubusercontent.com/117980596/215772121-f59db6da-25f8-46bb-ba09-7498eba6ced3.png)
.

**Comment by Beormund** [comment]
Bear in mind that there are TWO variables holding target temperature. If a zone is in AUTO mode and a schedule is active then self.target_temp is set to the schedule's target temperature. When a zone is first initialized, target_temp is set to zone['t']. zone.target is a static field name for zone['t']. zone.target_temp will change depending on whether or not a schedule is active which has it's own target temperature. 

https://github.com/Beormund/Tasmota32-Multi-Zone-Heating-Controller/blob/058a4a4e31b9d54358e03110ed7291b5d9a9788b/heating.be#L334

There may be a better way of doing this. If you can think of one please, by all means, suggest it or submit a PR!
